python/agents/gemini-fullstack/frontend/src/components/ActivityTimeline.tsx

```
@@ -0,0 +1,244 @@
+  const formatEventData = (data: any): string => {
+    // Handle new structured data types
+    if (typeof data === "object" && data !== null && data.type) {
+      switch (data.type) {
+        case 'functionCall':
+          return `Calling function: ${data.name}\nArguments: ${JSON.stringify(data.args, null, 2)}`;
+        case 'functionResponse':
+          return `Function ${data.name} response:\n${JSON.stringify(data.response, null, 2)}`;
+        case 'text':
+          return data.content;
+        case 'sources':
+          const sources = data.content as Record<string, { title: string; url: string }>;
```

Biome (1.9.4)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Unsafe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)